src/model/config.ts

```diff
@@ -0,0 +1,54 @@
+import BN from "bn.js";
+import { ConfigurableCorePool } from "../core/configurablePool";
+
+export class PoolConfig {
+    id: string;
+    tickSpacing: number;
+    token0: string;
+    token1: string;
+    fee: number;
+}
+
+export class Record {
+    id: string;
+    eventId: number;
+    actionName: string;
+    actionParams: object;
+    timestamp: Date;
+}
+
+export class Snapshot {
+    id: string;
+    nonce: number;
+    baseSnapshot: Snapshot;
```

Comment:
If the `Snapshot` is not incremental, and it already contains all the records, we probably do not need this field then?

Reply:
This is because I want to track how a snapshot came to state it is now from initial deployment action.

Reply:
I see what you are saying, maybe do something like this:

1. Add another class called `PoolState` which represents the current state of the pool.

2. Keep the class `Snapshot`, but flatten it so it contains everything in the `SnapshotStorage`, and remove class `SnapshotStorage`.

3. In the `PoolState`, it will have `baseSnapshot`  to keep track of which snapshot that the current state is recovered from, it will also have a list of `records` (deltas) that have been applied on top of the `baseSnapshot` to achieve the current state.

In short, `baseSnapshot` + `records` = `PoolState`

Reply:
I like this solution. Thank you.